flatfs.go

```diff
@@ -38,37 +40,52 @@ var _ datastore.Datastore = (*Datastore)(nil)
 
 type ShardFunc func(string) string
 
-func New(path string, getDir ShardFunc, sync bool) (*Datastore, error) {
-	fs := &Datastore{
-		path:   path,
-		getDir: getDir,
-		sync:   sync,
+func New(path string, fun0 string, sync bool) (*Datastore, error) {
+	if fun0 != "auto" {
+		fun0 = NormalizeShardFunc(fun0)
 	}
-	return fs, nil
-}
 
-func Prefix(prefixLen int) ShardFunc {
-	padding := strings.Repeat("_", prefixLen)
-	return func(noslash string) string {
-		return (noslash + padding)[:prefixLen]
+	fun1, err := ReadShardFunc(path)
+	if err != nil {
+		return nil, err
 	}
-}
 
-func Suffix(suffixLen int) ShardFunc {
-	padding := strings.Repeat("_", suffixLen)
-	return func(noslash string) string {
-		str := padding + noslash
-		return str[len(str)-suffixLen:]
+	fun := ""
+	if fun0 == "auto" && fun1 == "auto" {
```

Comment:
Okay.  I don't particular see the difference, but don't object either.  In the future I will try to remember that switch is generally preferred over multiple `else if`s.